newIDE/app/src/MainFrame/index.js

```diff
@@ -1071,249 +1086,254 @@ class MainFrame extends React.Component<Props, State> {
-              onOpenLanguageDialog={() => this.openLanguage()}
-              isActive={isActive}
-              ref={editorRef}
-            />
-          ),
-          key: 'start page',
-          closable: false,
-        }),
-      },
-      () => this.updateToolbar()
-    );
+    const { i18n, eventsFunctionsExtensionsState } = props;
+    setState(state => ({
+      ...state,
+      editorTabs: openEditorTab(state.editorTabs, {
+        label: name + ' ' + i18n._(t`(Extension)`),
+        renderEditor: ({ isActive, editorRef, newState, newProps }) => (
+          <EventsFunctionsExtensionEditor
+            project={newState.currentProject}
+            eventsFunctionsExtensionName={name}
+            setToolbar={setEditorToolbar}
+            resourceSources={newProps.resourceSources}
+            onChooseResource={(sourceName, multiSelection) =>
+              _onChooseResource(sourceName, multiSelection, newState)
+            }
+            resourceExternalEditors={newProps.resourceExternalEditors}
+            isActive={isActive}
+            initiallyFocusedFunctionName={initiallyFocusedFunctionName}
+            initiallyFocusedBehaviorName={initiallyFocusedBehaviorName}
+            openInstructionOrExpression={(extension, type) =>
+              _openInstructionOrExpression(extension, type, newState)
+            }
+            onCreateEventsFunction={(extensionName, eventsFunction) =>
+              _onCreateEventsFunction(extensionName, eventsFunction, newState)
+            }
+            ref={editorRef}
+            onLoadEventsFunctionsExtensions={() => {
+              eventsFunctionsExtensionsState.loadProjectEventsFunctionsExtensions(
+                newState.currentProject
+              );
+            }}
+            unsavedChanges={newProps.unsavedChanges}
+          />
+        ),
+        key: 'events functions extension ' + name,
+      }),
+    }));
+    openProjectManager(false);
+  };
+
+  const openResources = () => {
+    const { i18n } = props;
+    setState(state => ({
+      ...state,
+      editorTabs: openEditorTab(state.editorTabs, {
+        label: i18n._(t`Resources`),
+        renderEditor: ({ isActive, editorRef, newState, newProps }) => (
+          <ResourcesEditor
+            project={newState.currentProject}
+            setToolbar={setEditorToolbar}
+            onDeleteResource={(resource: gdResource, cb: boolean => void) => {
+              // TODO: Project wide refactoring of objects/events using the resource
+              cb(true);
+            }}
+            onRenameResource={(
+              resource: gdResource,
+              newName: string,
+              cb: boolean => void
+            ) => {
+              // TODO: Project wide refactoring of objects/events using the resource
+              cb(true);
+            }}
+            isActive={isActive}
+            ref={editorRef}
+            onChooseResource={(sourceName, multiSelection, newState) =>
```

Comment:
newState should not be in the argument of this callback, this will probably be undefined and so pass undefined to _onChooseResource

Reply:
This is an obvious mistake 😅 I think I messed up something with find and replace, sorry about that :D